albumentations/core/transforms_interface.py

```diff
@@ -38,6 +53,166 @@
 FillValueType = Optional[Union[int, float, Sequence[int], Sequence[float]]]
 
 
+@dataclass
+class BatchInternalType:
+    array: np.ndarray
+
+    def __setattr__(self, key, value):
+        if key == "array":
+            self.assert_array_format(value)
+        super().__setattr__(key, value)
+
+    @abstractmethod
+    def __getitem__(self, item):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __setitem__(self, key, value):
+        raise NotImplementedError
+
+    @abstractmethod
+    def check_consistency(self):
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def assert_array_format(array):
+        raise NotImplementedError
+
+
+@dataclass
+class BBoxesInternalType(BatchInternalType):
+    array: BoxesArray = field(default=np.empty((0, 4)))
+    targets: List[Any] = field(default_factory=list)
+
+    def __post_init__(self):
+        if not isinstance(self.array, np.ndarray):
+            self.array = np.array(self.array, dtype=float)
+        elif isinstance(self.array, np.ndarray):
+            self.array = self.array.astype(float)
+        if len(self.array) and not self.targets:
+            self.targets = [()] * len(self.array)
+        self.check_consistency()
+
+    def __eq__(self, other):
+        if not isinstance(other, BBoxesInternalType):
+            return False
```

Comment:
I thought it'll be better to be in this way?
just like when we compare [1,2 , 3] == (1, 2, 3), since they're in different type, we normally dont expected the program to raise error.

Reply:
unless we don't want others miss use this object in any cases?

Reply:
Yes, I think we should not compare this object with other types.